bootstrap.sh

```diff
@@ -1,39 +1,505 @@
+    } || {
+        (python <(wget -O- https://bootstrap.pypa.io/get-pip.py 2>/dev/null)) > /dev/null
+    }
+    fi
+    {
+        pip install --prefix /usr/local ansible
+    } || {
+        pip3 install --prefix /usr/local ansible
+    }
+    if [[ -z "$(command -v ansible)" ]]; then
+        help_install_ansible
+        return 1
+    else
+        echoSuccess "ansible installed!"
+        ansible --version
+        echo "" 
+        return 0
+    fi
+}
+
+check_ansible() {
+    {
+        if ! [[ -x "$(command -v ansible)" ]]; then
+            if [[ "$OSTYPE" == "linux-gnu" ]]; then
+                install_ansible "linux"
+            elif [[ "$OSTYPE" == "darwin"* ]]; then
+                # Mac OSX
+                install_ansible "darwin"
+            elif [[ "$OSTYPE" == "cygwin" ]]; then
+                # POSIX compatibility layer and Linux environment emulation for Windows
+                install_ansible "windows"
+            elif [[ "$OSTYPE" == "msys" ]]; then
+                # Lightweight shell and GNU utilities compiled for Windows (part of MinGW)
+                install_ansible "windows"
+            elif [[ "$OSTYPE" == "win32" ]]; then
+                # I'm not sure this can happen.
+                install_ansible "windows"
+            elif [[ "$OSTYPE" == "freebsd"* ]]; then
+                install_ansible "linux"
+            else
+                help_install_ansible
+                return 1
+            fi
+            return $?
+        else
+            echoSuccess "ansible found in path!"
+            ansible --version    
+            echo ""
+            return 0
+        fi
+    } || {
+        help_install_ansible
+        return 1
+    }
+}
+
+display_gcp_final_instructions() {
+    prettyTitle "Next Steps"
+    echo "Please run the following commands and add them in your shell profile file:"
+    echo "source $LIB_LOCATION/google-cloud-sdk/completion.bash.inc"
+    echo "source $LIB_LOCATION/google-cloud-sdk/path.bash.inc"
     echo ""
+}
+
+prettyHeader "Bootstrapping ioFog platform dependencies"
+
+if ! [[ -x "$(command -v curl)" ]]; then
+    echoError "curl not found"
+    exit 1
 fi
 
-# Terraform
-if [[ -z $(command -v terraform) ]] ; then
-    sudo apt-get update -qq && sudo apt-get install -qqy unzip
-    curl -fSL -o terraform.zip https://releases.hashicorp.com/terraform/"$TERRAFORM_VERSION"/terraform_"$TERRAFORM_VERSION"_"$OS"_amd64.zip
-    sudo mkdir -p /usr/local/opt/
-    sudo unzip -q terraform.zip -d /usr/local/opt/terraform
-    rm -f terraform.zip
-    sudo ln -s /usr/local/opt/terraform/terraform /usr/local/bin/terraform
+
+if [[ "$OSTYPE" == "cygwin" ]] || [[ "$OSTYPE" == "msys" ]] || [[ "$OSTYPE" == "win32" ]]; then
+    echoError "We do not currently support windows"
+    exit 1
+fi
+
+check_gcp
+gcp_success=$?
+check_tf
+tf_success=$?
+check_ansible
+ansible_success=$?
+check_kubectl
+kubectl_success=$?
+check_iofogctl
+iofogctl_success=$?
+
```

Comment:
would it be appropriate to have a --verify mode of bootstrap, that doesn't install anything at all, just checks if we have sufficient versions?

Reply:
Can be done.
There is no version validation step at the moment.